backend/api/v1/issue_service.go

```diff
@@ -1397,11 +1394,9 @@ func (s *IssueService) UpdateIssue(ctx context.Context, request *v1pb.UpdateIssu
 			}
 
 			if patch.PayloadUpsert == nil {
-				patch.PayloadUpsert = &storepb.IssuePayload{}
-			}
-			patch.PayloadUpsert.Approval = &storepb.IssuePayloadApproval{
-				ApprovalFindingDone: false,
+				patch.PayloadUpsert = payload
 			}
+			patch.PayloadUpsert.Approval.ApprovalFindingDone = request.Issue.ApprovalFindingDone
 
```

Comment:
payloadUpsert updates present top-level objects. So we don't want to set `patch.PayloadUpsert` to `payload` here.

Reply:
updated